core/src/services/gcs/core.rs

```diff
@@ -89,14 +91,18 @@ impl GcsCore {
         }
     }
 
-    fn load_credential(&self) -> Result<GoogleCredential> {
+    fn load_credential(&self) -> Result<Option<GoogleCredential>> {
+        if self.allow_anonymous {
```

Comment:
Hi, sorry for not making this clear before.

The meaning of `allow_anonymous` differs from `no_authentication`. We will still attempt to load credentials from the environment but will return `Ok(())` if no credentials are found.

Reply:
By default, we will attempt to load credentials from the environment if not provided through any supported methods and will return an error if they are not found. Based on this behavior, we have:

- `allow_anonymous` if credentials are not found
- `disable_abc` to disable the methods for loading credentials

This design ensures that all configuration settings are additive and work without overlaps.

Reply:
> sorry for not making this clear before

Not at all, I misunderstood.

This makes sense to me, I'll alter this :+1: